runtime/lua/vim/treesitter.lua

```diff
@@ -114,4 +114,106 @@ function M.get_string_parser(str, lang, opts)
   return LanguageTree.new(str, lang, opts)
 end
 
+--- Determines whether a node is the ancestor of another
+---
+---@param dest the possible ancestor
+---@param source the possible descendant node
+---
+---@returns True if dest is an ancestor of source
+function M.is_ancestor(dest, source)
+  if not (dest and source) then
+    return false
+  end
+
+  local current = source
+  while current ~= nil do
+    if current == dest then
+      return true
+    end
+
+    current = current:parent()
+  end
+
+  return false
+end
+
+--- Get the next node of the same parent
+---
+---@param node The start node
+---@param opts table A table of optional arguments:
+---                  - switch_parent: (boolean, default false) Allow to return the first node from
+---                  the next parent if the start node is the last node of its parent.
+---                  - return_next_parent: (boolean, default false) With switch_parent = true, allow
+---                  to return the next parent as the next node if it has no children.
+---
+---@returns The next node
+function M.get_next_node(node, opts)
```

Comment:
is this the same as `node:next_sibling()` when opts are not set? could be mentioned in the docs, as it is more efficient then.

Reply:
...or consolidated?